### DISCUSSION TITLE
AttributeError: Module 'tweepy' has no attribute 'Stream'

### DISCUSSION THREAD

**Discussion by zoiba-zia** [tweepy/tweepy#2136] [Questions] [answered]
I downloaded tweepy using pip. I am unable to import 'Stream' or use attribute 'Stream' with tweepy. I checked the module and __init__.py it doesn't have 'Stream" or 'StreamListener'. What do I do in this case?

This is my code: 
```
import tweepy
from tweepy import OAuthHandler
import json
import sentiment_mod as s

#consumer key, consumer secret, access token, access secret.
ckey="xyz"
csecret="xyz"
atoken="xyz"
asecret="xyz"

class listener(tweepy.stream):

    def on_data(self, data):

        all_data = json.loads(data)

        tweet = all_data["text"]
        sentiment_value, confidence = s.sentiment(tweet)
        print(tweet, sentiment_value, confidence)

        if confidence*100 >= 80:
            output = open("twitter-out.txt","a")
            output.write(sentiment_value)
            output.write('\n')
            output.close()

        return True

    def on_error(self, status):
        print(status)
stream = tweepy.stream(ckey="xyz", 
csecret="xyz",
atoken="xyz",
asecret="xyz")

auth = OAuthHandler(ckey, csecret)
auth.set_access_token(atoken, asecret)

twitterStream = Stream(auth, listener())
twitterStream.filter(track=["car"])

Solutions I have tried so far:
Reinstalling Tweepy
using attribute 'Stream'
creating a class 'StreamListener'
```

**Comment by Harmon758** [comment, marked as answer]
Yes, `StreamListener` was merged into `Stream` in [Tweepy v4.0.0](https://tweepy.readthedocs.io/en/v4.14.0/changelog.html#version-4-0-0-2021-09-25), almost two years ago, and `Stream` was removed in [Tweepy v4.14.0](https://tweepy.readthedocs.io/en/v4.14.0/changelog.html#version-4-14-0-2023-04-24), as streaming with Twitter API v1.1 has been deprecated.

`tweepy.Stream` was a class, not an attribute, and `tweepy.stream` never existed.

If you want to use streaming, you'll have to [use Twitter API v2 with `StreamingClient`](https://tweepy.readthedocs.io/en/v4.14.0/streaming.html#using-streamingclient) and Pro or Enterprise access.

For code block usage, see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.
